proxy.ts

```diff
@@ -0,0 +1,107 @@
+/**
+ * @internal Symbol used to track if the object has been mutated
+ */
+export const _MUTATED = Symbol("mutated");
+
+/**
+ * @internal Symbol used to track if the object is being tracked
+ */
+export const _TRACKED = Symbol("tracked");
+
+type Mutatable<T> = T & { [_MUTATED]?: boolean; [_TRACKED]?: boolean };
+
+// for Set, Map, and Array
+const MUTATION_METHODS = new Set([
+    "set",
+    "add",
+    "delete",
+    "clear",
+    "push",
+    "pop",
+    "shift",
+    "unshift",
+    "splice",
+    "sort",
+    "reverse",
+    "fill",
+    "copyWithin",
+]);
+
+/**
+ * @internal Creates a proxy wrapper around an object to track mutations
+ * @param object - The target object to proxify
+ * @returns Proxied object that tracks mutations
+ */
+export function trackMutations<T>(
+    object: T,
+): Mutatable<T> {
+    if (typeof object !== "object" || object === null) {
+        return object as Mutatable<T>;
+    }
```

Comment:
So … strings are returned as-is? Is this correct?

Reply:
It is. There's nothing to track besides direct assignment and that's done on the parent object. Also:
![CleanShot 2025-04-06 at 09 35 20@2x](https://github.com/user-attachments/assets/18476a92-6dae-4bdf-ba15-edd89137ae83)